src/wallet/wallet.cpp

```diff
@@ -3323,7 +3323,7 @@ bool CWallet::InitLoadWallet()
 
     RegisterValidationInterface(walletInstance);
 
-    CBlockIndex *pindexRescan = chainActive.Tip();
+    CBlockIndex *pindexRescan = nullptr;
     if (GetBoolArg("-rescan", false))
```

Comment:
this looks to me like you're changing the functionality of rescan and will cause more rescans on startup than is necessary.

Reply:
Hmm? As far as I can see, `pindexRescan` is initialized in all codepaths below and thus this one seems superfluous - and was detected by clang SA as such.

Or are you intending a side-effect of calling `chainActive.Tip()` here? As far as I can see, that method also simply returns a value.

Reply:
I was looking at line 3338.  But you're right, pindexRescan by the time it gets to 3338 will have been assigned another value so it's fine.